onnx/defs/nn/defs.cc

```diff
@@ -1248,6 +1256,44 @@ void convTransposeShapeInference(InferenceContext& ctx) {
     }
   }
 
+  std::vector<int64_t> pads;
+  if (getRepeatedAttribute(ctx, "pads", pads)) {
+    if (pads.size() != n_input_dims * 2) {
```

Comment:
According to the spec, it is totally possible for pads size to be less than 2 * input dimensions. In this case, default values to 0. Probably instead of failing shape inference, you could do a resize with values set to 0. This is the line in the spec which makes it possible - 

`If not present, the padding defaults to 0 along start and end of each axis.`

Reply:
I checked the spec again. My understanding is that if "pads" presents, its size should be n_input_dims*2. If it doesn't present, its default value is 0, but its size should still be n_input_dims*2.